libs/wingc/src/lib.rs

```diff
@@ -273,15 +274,49 @@ pub fn type_check(
 		types,
 	);
 
-	let mut tc = TypeChecker::new(types, file_path, file_graph, jsii_types, jsii_imports);
-	tc.add_jsii_module_to_env(
+	add_jsii_module_to_env(
 		&mut env,
 		WINGSDK_ASSEMBLY_NAME.to_string(),
 		vec![WINGSDK_STD_MODULE.to_string()],
 		&Symbol::global(WINGSDK_STD_MODULE),
 		None,
+		file_path,
+		types,
+		jsii_imports,
+		jsii_types,
 	);
 
+	let std_node_fqn = format!("{}.{}", WINGSDK_ASSEMBLY_NAME, WINGSDK_NODE);
+	let std_node = types
+		.libraries
+		.lookup_nested_str(&std_node_fqn, None)
+		.expect("std.Node not found in type system")
+		.0
+		.as_type()
+		.expect("std.Node was found but it's not a type");
+
+	add_builtin(
+		UtilityFunctions::Nodeof.to_string().as_str(),
+		// important: the function type must be the same as std.Node.of
+		Type::Function(FunctionSignature {
+			this_type: None,
+			parameters: vec![FunctionParameter {
+				name: "construct".into(),
+				typeref: types.construct_interface(),
+				docs: Docs::with_summary("The construct to obtain the tree node of"),
+				variadic: false,
+			}],
+			return_type: std_node,
+			phase: Phase::Preflight,
+			js_override: None, // implementation replaced by nodeof_transform
+			docs: Docs::with_summary("Obtain the tree node of a preflight resource."),
+		}),
+		scope,
+		types,
+	);
```

Comment:
Instead of doing this here and refactoring `add_jsii_module_to_env`, would it be reasonable to put this `add_builtin` with the others in the typechecker and then do the transform after tc is complete?

Reply:
Done 👍